pkg/segment/writer/segstore.go

```diff
@@ -825,6 +827,10 @@ func (segstore *SegStore) checkAndRotateColFiles(streamid string, forceRotate bo
 		updateRecentlyRotatedSegmentFiles(segstore.SegmentKey)
 		metadata.AddSegMetaToMetadata(&segmeta)
 
+		// TODO: make this a background job; somehow handle siglens being
+		// gracefully shutdown.
+		writeSortIndexes(segstore.SegmentKey)
```

Comment:
for now should we just make this a goroutine, atleast the seg rotation will not be stuck here waiting for the sorted index to be created ?

Reply:
I'd slightly prefer to keep it as is until it's properly fixed. This function is a no-op unless the sort index feature flag is enabled, btw.

The downside of keeping this as is is that ingestion stalls for a little after rotation, like you said. The downside of changing this to a goroutine without doing the proper fix, is that if someone gracefully shuts down and that causes segment rotation, that could cause a .srt file to be only half written. When siglens is restarted, queries sorting on that column will give bad results if they need data from the unwritten part of that .srt file.

Either way, there's no downside when the feature flag is disabled

Reply:
sounds good